script.js

```diff
@@ -1,4 +1,388 @@
+      sortedList = [...list];
+
+      finalScoreboard = finalScoreboardStatement(
+        selectionSortForScoreboard(sortedList, gameMode)
+      );
+
+      var output = `Welcome Player ${playerNum} <br> ${playeraction}. <br><br> ${sortedList[0].player} is the winner!<br>${finalScoreboard} `;
+      playerNum++;
+      if (playerNum - 1 == numberOfPlayers) {
+        console.log("read");
+        playerNum = 1;
+        firstRound = false;
+      }
+      return output;
+    }
+    //running knockout gamemode
+    if (gameMode == "knockout") {
+      var scoreBoard = {
+        player: `Player ${playerNum}`,
+        score: combinedScoring,
+      };
+      console.log(scoreBoard);
+      list.push(scoreBoard);
+
+      console.log(list[0].score);
+      //initate the first round
+      if (playerNum == 1) {
+        playerNum++;
+        return `${list[0].player} is the winner. <br> ${playeraction}. <br><br> Your Score is ${list[0].score}<br> It is player ${playerNum}'s turn.`;
+      }
+      console.log(playerNum);
+      //compare if second player is higher than first player
+      if (list[1].score > list[0].score) {
+        playerNum++;
+        if (playerNum > numberOfPlayers) {
+          var insert = "<br>The game has ended.";
+        } else {
+          var insert = `<br> It is player ${playerNum}'s turn.`;
+        }
+
+        var output = `You are ${list[1].player}.<br><br>${list[1].player} is the winner. <br> ${playeraction}. <br><br> ${list[1].player} score is ${list[1].score}<br>${list[0].player} score is ${list[0].score}.`;
+        list[0] = list[1];
+        list.pop(scoreBoard);
+        console.log(list);
+
+        return output + insert;
+      }
+      //compare if first player is higher than second player.
+      if (list[0].score > list[1].score) {
+        playerNum++;
+        if (playerNum > numberOfPlayers) {
+          var insert = "<br>The game has ended.";
+        } else {
+          var insert = `<br> It is player ${playerNum}'s turn.`;
+        }
+        var output = `You are ${list[1].player}.<br><br> ${list[0].player} is the winner. <br> ${playeraction}. <br><br> ${list[0].player} score is ${list[0].score}<br>${list[1].player} score is ${list[1].score}.`;
+        list.pop(scoreBoard);
+
+        return output + insert;
+      }
+
+      if (playerNum - 1 == numberOfPlayers) {
+        return `The knockout game has ended <br> ${list[0].player} is the overall winner with a score of ${list[0].score}. `;
+      } else return "Roll again.";
+    }
+  }
+
+  //Play game in reversed gamemode
+  if (
+    gameMode == "reversed" &&
+    gameModechosen == true &&
+    NumberOfPlayersChosen == true
+  ) {
+    if (input == 0 || input == 1) {
+      input = 2;
+    }
+    diceResults = rollTheDice(input);
+    lowestCombinedNumber = selectionSort(diceResults, gameMode);
+    combinedScoring = combinedScore(lowestCombinedNumber);
+    playeraction = playerAction(diceResults, input, combinedScoring, gameMode);
+
```

Comment:
As these are applicable for the first game modes too, might want to put this on top before the code is separated for normal and reversed game mode